packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -364,12 +321,23 @@ function parseChildren(
             } else if (parse5Utils.isTextNode(child)) {
                 const textNodes = parseText(ctx, child);
                 parent.children.push(...textNodes);
+                // Non whitespace text nodes interrupt any context we may be carrying
+                if (textNodes.length > 0) {
+                    ctx.clearSiblingScope();
```

Comment:
IMO for this block and one below with comments, we do something like `ctx.visitTextNode` or `ctx.visitCommentNode` and let the logic for when to clear the sibling scope live inside the context object. Any thoughts on naming?